runtime/vm/JVMImage.cpp

```diff
@@ -283,28 +288,29 @@ JVMImage::registerEntryInTable(ImageTableHeader *table, UDATA entry)
 }
 
 bool
-JVMImage::readImageFromFile()
+JVMImage::readImageFromFile(void)
 {
 	Trc_VM_ReadImageFromFile_Entry(_heap, _dumpFileName);
 
 	OMRPORT_ACCESS_FROM_OMRPORT(getPortLibrary());
 
 	intptr_t fileDescriptor = omrfile_open(_dumpFileName, EsOpenRead, 0444);
-	if (fileDescriptor == -1) {
+	if (-1 == fileDescriptor) {
 		return false;
 	}
 
-	// Read image header then mmap the rest of the image (heap) into memory
-	JVMImageHeader imageHeader;
-	omrfile_read(fileDescriptor, &imageHeader, sizeof(JVMImageHeader));
+	/* Read image header then mmap the rest of the image (heap) into memory */
+	omrfile_read(fileDescriptor, (void *)_jvmImageHeader, sizeof(JVMImageHeader));
 	uint64_t fileSize = omrfile_flength(fileDescriptor);
-	if (fileSize != sizeof(JVMImageHeader) + imageHeader.imageSize) {
+	if (sizeof(JVMImageHeader) + _jvmImageHeader->imageSize != fileSize) {
```

Comment:
So for this image size is already defined as sizeof(JVMImageHeader) + heapSize; imageSize =/= heapSize

Reply:
Ooooops
Will fix, I will also fix the write function so just makes one "write" function call.